### PR TITLE
disttask: remove mini executor abstraction

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

### What is changed and how it works?

remove mini executor abstraction. rename SplitSubtask to RunSubtask, and other refactor.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/stage_ingest_index.go [hunk 1]
```diff
@@ -61,9 +61,9 @@ func (i *ingestIndexStage) Init(ctx context.Context) error {
 	return err
 }
 
-func (*ingestIndexStage) SplitSubtask(ctx context.Context, _ *proto.Subtask) ([]proto.MinimalTask, error) {
```
Comment (ywqzzy) [inline ddl/stage_ingest_index.go:64]: Maybe we can move code in init function to RunSubtask

Comment (D3Hunter) [reply ddl/stage_ingest_index.go:64]: init is done once for all subtasks, RunSubtask is per subtask

**Review by lance6716 — APPROVED** [review]
excited!

**Review by tangenta — APPROVED** [review]

**Pull request merged**